tests/photo/test_phz.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import pytest
+
+import euclidlib as el
+
+
+@pytest.mark.parametrize("read_hist", [True, False])
+@pytest.mark.parametrize("write_hist", [True, False])
+def test_redshift_distributions(tmp_path, write_hist, read_hist):
+    def fn(z, hist):
+        """produce a mock n(z), optionally binned into a histogram"""
+        nz = np.exp(-((z - 1.0) ** 2) / (0.5) ** 2 / 2)
+        if hist:
+            nz = (nz[:-1] + nz[1:]) / 2 * np.diff(z)
+        return nz
+
+    # write test data from higher redshift resolution
+    z = np.linspace(0.0, 6.0, 10001)
+    nz = fn(z, write_hist)
+
+    path = tmp_path / "nz.fits"
+
+    el.photo._phz.write_nz(path, z, nz, hist=write_hist)
+
+    # read test data
+
+    z, nz = el.photo.redshift_distributions(path, hist=read_hist)
+
+    np.testing.assert_array_equal(z, np.linspace(0.0, 6.0, 3001))
+
+    assert len(nz) == 1
+    assert list(nz.keys()) == [1]
+
+    nz_ = fn(z, read_hist)
+    np.testing.assert_allclose(nz[1], nz_, rtol=1e-2, atol=1e-4)
+
+
+def test_redshift_distributions_ident(tmp_path):
+    """
+    Test that writing a histogram in the correct format returns the data
+    unchanged.
+    """
+    z_ = np.linspace(0.0, 6.0, 3001)
```

Comment:
I am still not happy about having this hardcoded, even if it comes from the SGS and it's fixed. We can open an issue to remember to discuss this in the future.

Reply:
The binning is not hardcoded in what write_nz() accepts, it's only hardcoded in the the FITS file. This test makes sure that if you **do** give the writer a histogram in the PHZ format, then it is read and written unchanged.

Reply:
Just for me to understand: what happens if this particular selection is changed in the future by the PHZ people? For the writing routine (even if we keep it private), we would need to modify it, right?

Reply:
That's right. But the same is true for every other aspect of the format, e.g. the FITS keywords or extension names (both changed recently).